src/workerd/api/http.c++

```diff
@@ -1887,7 +1887,13 @@ jsg::Promise<jsg::Ref<Response>> fetchImplNoOutputLock(
       nativeRequest = client->request(jsRequest->getMethodEnum(), url, headers, uint64_t(0));
     }
     return ioContext.awaitIo(js,
-        AbortSignal::maybeCancelWrap(signal, kj::mv(KJ_ASSERT_NONNULL(nativeRequest).response)),
+        AbortSignal::maybeCancelWrap(signal, kj::mv(KJ_ASSERT_NONNULL(nativeRequest).response))
+            .catch_([](kj::Exception&& exception) -> kj::Promise<kj::HttpClient::Response> {
+          if (exception.getDescription().startsWith("invalid Content-Length header value")) {
+            return JSG_KJ_EXCEPTION(FAILED, Error, exception.getDescription());
```

Comment:
@jasnell Minor point, but JavaScript error message style is usually more formal than KJ error message style, e.g. expecting complete sentences with a capitalized first letter and terminating punctuation.

On another note, I don't think it's possible for this error to happen in production since the invalid value would never get past FL.